### DISCUSSION TITLE
Yet another contrast mask method: Difference of statistic

### DISCUSSION THREAD

**Discussion by mviereck** [ImageMagick/ImageMagick#5736] [Digital Image Processing] [answered]
I just want to share another possibility to create contrast masks.
The idea is to compare the results of two `-statistic` results.
Most of interest is the comparison of mean|median and of minimum|maximum.

Examples:
```
magick \( wizard: -grayscale RMS -statistic mean 2 \)  \
        \( wizard: -grayscale RMS -statistic median 2 \)  \
        -compose Difference -composite -auto-level x:
```
![dos_meanmedian](https://user-images.githubusercontent.com/14821372/200399269-3758b22d-4ece-4ac4-9b2c-a101211527f9.png)

```
magick \( wizard: -grayscale RMS -statistic minimum 2 \)  \
        \( wizard: -grayscale RMS -statistic maximum 2 \)  \
        -compose Difference -composite -auto-level x:
```
![dos_minmax](https://user-images.githubusercontent.com/14821372/200399177-0e6347be-e6c6-4f0c-bdb8-a40f730f4719.png)

Maybe this is of interest for you.
I could even imagine an imagemagick convolution option `DoS` similar to `-morphology Convolve DoG`.
Or the method could be just mentioned in some contrast mask documentation.

**Comment by urban-warrior** [comment, marked as answer]
The principles of orthogonality suggest that there should only be one way to do one thing.  Given you have provided a command-line to accomplish your goals, how would adding a new contrast method differentiate from what you have illustrated here?  Is it just a matter of convenience?

Another reason for introducing a new feature is if we can optimize the performance.  Reviewing your command-line, we do not spot any opportunities for increased performance by consolidating into a single contrast method.

The user community is now aware of your contrast algorithm suggestion per this post.  Perhaps that is sufficient.

**Reply by mviereck** [reply]
If there is no performance to gain in adding a new option, then there is no good reason to add one.
Somehow documenting this possibility would be sufficient.
My main intention was to just share this neat possibility for contrast mask generation.

As snibgo pointed out, `-statistic gradient` is already given for min/max comparision.
An additional argument for `-statistic` to compare median/mean might make sense. Not sure how it should be called.

**Comment by snibgo** [comment]
The difference between the maximum and minimum is already available as a single operation: "`-statistic Gradient`". See https://imagemagick.org/script/command-line-options.php#statistic

**Reply by mviereck** [reply]
Indeed, thank you for pointing me on this!
